### PR TITLE
Add last trade to snapshot

### PR DESCRIPTION
**Pull request by kubilayluno**

### CODE REVIEW

File: streaming/streaming.go [hunk 1]
```diff
@@ -345,8 +346,16 @@ func (c *Conn) processTrade(t TradeUpdate) error {
 		return errors.New("streaming: nonpositive trade")
 	}
 
+	last := c.lastTrade
```
Comment (neilgarb) [inline streaming/streaming.go]: I actually don't think rolling back lastTrade is necessary. I would just do:

```
 c.lastTrade = TradeUpdate{Base: t.Base, Counter: t.Counter}
```

here and that's it.

Comment (neilgarb) [reply streaming/streaming.go]: Also I don't think we need the order ID. We're dropping one order ID or the other anyway - might as well not include either.

File: streaming/streaming_internal_test.go [hunk 2]
```diff
@@ -370,13 +377,32 @@ func existsInOtherMap(a, b map[string]order) bool {
...
+	if &a == &b {
+		return true
+	}
+	if a.Base.Cmp(b.Base) != 0 {
+		return false
+	}
+	if a.Counter.Cmp(b.Counter) != 0 {
+		return false
+	}
+	if a.OrderID != b.OrderID {
```
Comment (neilgarb) [inline streaming/streaming_internal_test.go]: Can drop the order ID check.

File: streaming/streaming_internal_test.go [hunk 3]
```diff
@@ -87,8 +88,8 @@ func TestReceivedUpdate(t *testing.T) {
 			args: args{
 				u: Update{Sequence: 2,
 					TradeUpdates: []*TradeUpdate{
-						{Base: decimal.NewFromFloat64(0.01, 2),
-							Counter: decimal.NewFromFloat64(0.001, 2), OrderID: "1"},
+						{Base: decimal.NewFromFloat64(0.02, 2),
```
Comment (neilgarb) [inline streaming/streaming_internal_test.go]: Please add a test case to check that if no updates are received, LastTrade is zero.

File: streaming/streaming.go (hunk 1 above)
Comment (kubilayluno) [reply streaming/streaming.go]: Not sure what if streaming process fails? Do you mean if streaming receives an update, this means transaction had already processed, streaming may not process it but it actually is the last trade and should update it any case?

Relating with not setting order ID: makes sense, no need to export it. Will amend it

File: streaming/streaming_internal_test.go (hunk 2 above)
Comment (kubilayluno) [reply streaming/streaming_internal_test.go]: Will do

File: streaming/streaming_internal_test.go (hunk 3 above)
Comment (kubilayluno) [reply streaming/streaming_internal_test.go]: TestOrderBook(), makes that check if I am not mistaken?
